addons/hr_expense/tests/test_expenses.py

```diff
@@ -1179,3 +1181,86 @@ def test_expense_from_attachments(self):
         self.env['hr.expense'].create_expense_from_attachments(attachment.id)
         expense = self.env['hr.expense'].search([], order='id desc', limit=1)
         self.assertEqual(expense.account_id, product.property_account_expense_id, "The expense account should be the default one of the product")
+
+    def test_expense_sheet_attachments_sync(self):
+        """
+        Test that the hr.expense.sheet attachments stay in sync with the attachments associated with the expense lines
+        Syncing should happen when:
+        - When adding/removing expense_line_ids on a hr.expense.sheet <-> changing sheet_id on an expense
+        - When deleting an expense that is associated with an hr.expense.sheet
+        - When adding/removing an attachment of an expense that is associated with an hr.expense.sheet
+        """
+        def assert_attachments_are_synced(attachments_from_expenses, attachments_from_sheet, attachments_on_sheet, sheet_has_attachment):
+            if sheet_has_attachment:
+                self.assertTrue(bool(attachments_on_sheet), "Attachment that belongs to the hr.expense.sheet only was removed unexpectedly")
+            self.assertSetEqual(set(attachments_from_expenses.mapped('checksum')), set((attachments_from_sheet - attachments_on_sheet).mapped('checksum')),
+                                "Attachments between expenses and their sheet is not in sync.")
+
+        for sheet_has_attachment in (False, True):
+            expense_with_attachment_1, expense_with_attachment_2, expense_with_attachment_3 = self.env['hr.expense'].create([{
+                'name': 'expense_1',
+                'employee_id': self.expense_employee.id,
+                'product_id': self.product_c.id,
+                'total_amount': 1000,
+            }, {
+                'name': 'expense_2',
+                'employee_id': self.expense_employee.id,
+                'product_id': self.product_c.id,
+                'total_amount': 999,
+            }, {
+                'name': 'expense_3',
+                'employee_id': self.expense_employee.id,
+                'product_id': self.product_c.id,
+                'total_amount': 998,
+            }])
+            self.env['ir.attachment'].create([{
+                'name': "test_file_1.png",
```

Comment:
let's avoid the annoying log `Post processing ignored : This file could not be decoded as an image file.`
by using a .txt file (or anything else actually, as long as the logs are not polluted :) )

Reply:
hmmm didn't see this line in my logs, odd, I've changed the extension to txt, should be ok?